src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,284 @@
+        new Duke().start();
+    }
+
+    /**
+     * Starts MakiBot.
+     */
+    protected void start() {
+        System.out.println("Hello! I'm MakiBot");
+        timeZone = ui.getTimeZone(timeZone);
+        storage.setSaveFilePath(ui.getSaveFile(storage.getSaveFilePath()));
+        tasks = new TaskList(storage.loadTasks(timeZone));
+        System.out.println("Welcome! What can I do for you?");
+        eventLoop();
+    }
+
+    /**
+     * Starts a conversation with MakiBot.
+     */
+    protected void eventLoop() {
+        while (true) {
+            try {
+                // Event loop
+                String input = ui.getInput();
+                String[] fullCommand = parser.parseFullCommand(input);
+                Command command = parser.parseCommand(input);
+
+                // Handle commands
+                switch (command) {
+                // Exit command
+                case BYE:
+                    storage.updateSaveFile(tasks.getTasks());
+                    ui.close();
+                    return;
+                case SAVE:
+                    storage.updateSaveFile(tasks.getTasks());
+                    break;
+                // List all tasks
+                case LIST:
+                    ui.printAllTasks(tasks.getTasks());
+                    break;
+                // Mark task as done
+                case MARK:
+                    mark(fullCommand);
+                    break;
+                // Mark task as undone
+                case UNMARK:
+                    unmark(fullCommand);
+                    break;
+                case DELETE:
+                    delete(fullCommand);
+                    break;
+                case FIND:
+                    find(fullCommand);
+                    break;
+                case TODO:
+                    newTodo(fullCommand);
+                    break;
+                case DEADLINE:
+                    newDeadline(fullCommand);
+                    break;
+                case EVENT:
+                    newEvent(fullCommand);
+                    break;
+                default:
+                    throw new DukeInvalidCommandException();
+                }
+            } catch (DukeException de) {
+                System.out.println(de.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Marks a task from the task list as done.
+     *
+     * @param fullCommand The full input from the user containing the task's index.
+     * @throws DukeException If the input is invalid.
+     */
+    protected void mark(String[] fullCommand) throws DukeException {
+        if (fullCommand.length < 2 || fullCommand[1].equals("")) {
+            throw new DukeFormatCommandException("mark");
+        }
+
+        try {
+            int taskNum = Integer.parseInt(fullCommand[1]) - 1;
+            Task t = tasks.get(taskNum);
+            t.markAsDone();
+            System.out.println("Nice! I've marked this task as done:\n" + t);
```

Comment:
Speaking for all similiar cases, perhaps these outputs could be handled by a class? like a ChatBot class for better encapsulation